Inventory Summary Result

nicocharlery opened the pull request:
This PR aims to request the inventory summary to Quiet Logistic, and with this result, update the corresponding variants count_on_hand for the right stock location.

I've taken the position of not creating or deleting variants out of the summary. A variant should be created first on Spree and have its count_on_hand updated.

nicocharlery committed:
Clean and Document Test Setup

nicocharlery committed:
Fix RMA Document Sending Test

The cassette had registered a specific number and date stamp. We needed then to
reproduce the same.

To debug this, I have used
`  c.debug_logger = File.open(ARGV.first + '.log', 'w')`
in spec/support/vcr.rb

nicocharlery committed:
Fix RMA Document Sending Test

The cassette had registered a specific number and date stamp. We needed then to
reproduce the same.

To debug this, I have used
`  c.debug_logger = File.open(ARGV.first + '.log', 'w')`
in spec/support/vcr.rb

nicocharlery committed:
Merge branch 'chore/clean_test_suite' into feature/inventory_summary

nicocharlery committed:
Send InventorySummary Request to SQS

nicocharlery committed:
Handle InventorySummary Results Without Reacting to it

This will handle the responses and errors after asking for an InventorySummary.
However, it does not yet update the count on hand on the variants.

nicocharlery committed:
Move Shipment Number from Spec to RMADecorateStub

The shipment number was not exercised in the spec. We then move it to the
RMADecorateStub.

nicocharlery committed:
Remove country_name Helper

`address.country_name` was a decoration present in the Aday project. We then use
the bare behaviour from Spree.

nicocharlery committed:
Set Count_on_hand Received From InventorySummaryResult

This will set the count_hand_hand for the right variant and stock_location.

nicocharlery committed:
Merge branch 'chore/clean_test_suite' into feature/inventory_summary

nicocharlery committed:
Lint

nicocharlery committed:
Add Warehouse Name As Argument to Request Inventory Summary

Since I don't know how is presented a document with results from different
warehouses, a safe approach is to ask a document per warehouse.

nicocharlery committed:
Bump Version

DanielWright commented on lib/ship_quiet_logistics/handlers/error_message_handler.rb:
```diff
@@ -4,7 +4,8 @@ class ErrorMessageHandler
         shipment = Spree::Shipment.find_by(number: message.shipment_number)
 
-        shipment.update(pushed: false)
+        shipment.update(pushed: false) if shipment
```
Should this fail silently if the shipment can't be found?

bokay replied:
The issue I got is that the error_message_handle is generic for every type of request:
https://github.com/DynamoMTL/ship_quiet_logistics/blob/master/lib/ship_quiet_logistics.rb#L45

Thus, in the case of an inventory summary request, there is no shipment to look up and it broke because of that. A better design I think would be to have a error handler per type of request, but I feel it out of scope for this PR.

DanielWright replied:
I don't think it's necessarily out of scope. The scope of this PR is handling inventory-summary documents, part of which is handling errors in those.

bokay replied:
What do you think of this proposal to manage different type of errors : https://github.com/DynamoMTL/ship_quiet_logistics/pull/10/commits/1af9f510e5b0dfb62613e95c8eff6e2d520f6776 ?

DanielWright reviewed (approved):
Looks really good!

nicocharlery committed:
Fix Each Skipping Keyword

Return might break the loop instead of skipping and going to the next item to
process.

nicocharlery committed:
Introduce Different Types of Error Handlers

Before this commit, we had one class managing all the errors for any kind of
message.
Now the ErrorMessageHandler purpose is to dispatch the error the a class that
know how to handle a specific error.

nicocharlery commented:
Out of date.
